test/unit-tests/type/unit/Unit.test.js

```diff
@@ -67,6 +67,21 @@ describe('Unit', function () {
       assert.strictEqual(unit1.units[0].unit.name, 'm3')
     })
 
+    it('should create a unit from an existing unit', function () {
+      const unit1 = new Unit(null, 'm/s^2')
+      const unit2 = new Unit(5, unit1)
+      assert.strictEqual(unit2.value, 5)
+      assert.strictEqual(unit2.units.length, 2)
+      assert.strictEqual(unit2.units[0].unit.name, 'm')
+      assert.strictEqual(unit2.units[1].unit.name, 's')
+      assert.strictEqual(unit2.units[1].power, -2)
+    })
+
+    it('should create a unitless Unit if second parameter is undefined', function () {
+      const a = new Unit(6)
+      assert(a.dimensions.every(d => d === 0))
+    })
+
```

Comment:
Is this a "public" behavior of the Unit class that we want to advertise and set in stone by adding tests for it, and potentially have users come to rely on? From my point of view, unitless Units are a troublesome thing (hence the proposal to separate into PureUnit and Quantity) because there's nothing really to distinguish them from raw numeric values. I see their existence as an implementation detail of Unit as it stands that could (and conceivably ought to) go away in a future revision. Certainly if the eventual external Unit library doesn't have such a thing, I wouldn't weep. Anyhow, open to your thoughts and @josdejong might want to weigh in.

Reply:
Having a unitless Unit will happen if you, for example, divide two Units which have the same dimensions. So I think this case should be allowed.

Reply:
I checked and the current behavior in mathjs is that this becomes a numeric value, not an instance of Unit, which definitely seems like the expected behavior to me...

Reply:
In Math.js, yes, I agree this should be the default behavior. But if I recall, using `predictable: true` will leave it as a Unit. This is also what UnitMath will do, so that users can chain operations and intermediate values won't be unexpectedly demoted to numbers.

Reply:
Right so it could well be the case that something like this could be a unit test for UnitMath, but it seems like an out of place test for mathjs which could get along fine with a unit implementation that had no such thing as a "unitless unit" if ever that was a thing like that. In other words, I don't see any purpose served by mathjs asserting that units have to work this way. That's all.

Reply:
It would simplify the Unit constructor by a few lines if we remove the ability to construct a unitless Unit.

Reply:
I wasn't actually proposing removing the ability either. Just not enshrining it in our tests, as it's not really part of the "public API" in my view. Just my two cents, we'll see if Jos has any thoughts.

Reply:
Sorry, just trying to maintain the compatibility. I have reverted the test.

Reply:
Thanks for adding more tests @costerwi. 

I think it is best to keep `Unit` fully backward compatible for now, until we switch to `UnitMath`). That means also keeping this existing, undocumented/unused behavior in place. Removing this undocumented functionality could break peoples code in subtle ways if they do "smart" things on top of `Unit`, therefore I think we should just keep this code as is. Even though it is not the "officially promoted" way of using `Unit`, I think it will not hurt to guard this behavior with a unit test. I prefer consciously making changes in the API (having to change tests) over keeping the behavior fuzzy, untested, and undocumented.

Reply:
(I think it's a shortcoming that I didn't create a test for that case in the first place when I created the constructor long time ago)